### PR TITLE
W-11365512: Use new API for array validation

### PR DESCRIPTION
**Pull request by dcourtil**

### CODE REVIEW

File: src/main/java/org/mule/module/apikit/validation/attributes/HeadersValidator.java [hunk 1]
```diff
@@ -164,8 +164,10 @@ private static List<String> deserializeValues(List<String> listOfDelimitedValues
   }
 
   private static void validateTypeArrayValues(String name, List<String> values, Parameter type) throws InvalidHeaderException {
-    String yamlArrayValue = values.stream().collect(Collectors.joining("\n- ", "- ", ""));
-    validateTypeValue(name, yamlArrayValue, type);
+    if (!type.validateArray(values)) {
+      throw new InvalidHeaderException(format("Invalid values '%s' for header '%s'", values.stream().collect(joining(",")),
```
Comment (iglosiggio) [inline src/main/java/org/mule/module/apikit/validation/attributes/HeadersValidator.java:168]: Probably a bit clearer:
```suggestion
      throw new InvalidHeaderException(format("Invalid values '%s' for header '%s'", join(",", values)),
```

In that case the static import for `java.util.stream.Collectors.joining` should be removed and the static import for `java.lang.String.join` should be added.

File: src/test/munit/headers/deserializing/header-array-default-deserializing-test-suite.xml [hunk 2]
```diff
@@ -607,7 +607,7 @@
         </munit:validation>
     </munit:test>
 
-    <munit:test name="success-including-quoted-datetime-array-header-default">
+    <munit:test name="fail-with-quoted-datetime-array-header-default">
```
Comment (iglosiggio) [inline src/test/munit/headers/deserializing/header-array-default-deserializing-test-suite.xml:610]: Can you explain why these tests changed from success to failure? Was the previous success a bug?

**Review by iglosiggio — APPROVED** [review]
The changes look good! I've just added a small recomendation and a question :)

File: src/test/munit/headers/deserializing/header-array-default-deserializing-test-suite.xml (hunk 2 above)
Comment (dcourtil) [reply src/test/munit/headers/deserializing/header-array-default-deserializing-test-suite.xml:610]: Correct, this is a (breaking) change to keep compatibility with the way the rest of the parameter values are validated. Quotes are not required to be added by the developer anymore, instead, they are considered to be part of the value. That's why the test has to fail (the date between quotes does not have a valid date format).
However, when the CSV deserializing strategy is enabled, quotes are required to delimit values that might have delimiters inside it, so quotes in that case are not considered to be part of the value.
Thanks for asking @iglosiggio, we can continue to discuss this with the rest of the team to agree if this is the behavior that we want, CCs @JuanAller @juanchib

**Pull request merged**